pcdsdevices/pseudopos.py

```diff
@@ -89,5 +92,75 @@ def inverse(self, real_pos):
         """
         Combined axis readback is the mean of the composite axes
         """
-        real_pos = self.RealPosition(*real_pos)
         return self.PseudoPosition(pseudo=self.calc_combined(real_pos))
+
+
+class DelayBase(PseudoPositioner):
+    """
+    Laser delay stage to rescale a physical axis to a time axis.
+
+    The optical laser travels along the motor's axis and bounces off a number
+    of mirrors, then continues to the destination. In this way, the path length
+    of the laser changes, which introduces a variable delay. This delay is a
+    simple multiplier based on the speed of light.
+
+    Attributes
+    ----------
+    delay: ``PseudoSingle``
+        The fake axis. It has configurable units and number of bounces.
+
+    motor: ``PositionerBase``
+        The real axis. This can be a number of things based on the inheriting
+        class, but it must have a valid ``egu`` so we know how to convert to
+        the time axis.
+
+    Parameters
+    ----------
+    prefix: ``str``
+        The EPICS prefix of the real motor
+
+    name: ``str``, required keyword
+        A name to assign to this delay stage.
+
+    egu: ``str``, optional
+        The units to use for the delay axis. The default is seconds. Any
+        time unit is acceptable.
+
+    n_bounces: ``int``, optional
+        The number of times the laser bounces on the delay stage, e.g. the
+        number of mirrors that this stage moves. The default is 2, a delay
+        branch that bounces the laser back along the axis it enters.
+    """
+    delay = FCpt(PseudoSingle, egu='{self.egu}', add_prefix=['egu'])
+    motor = None
+
+    def __init__(self, *args, egu='s', n_bounces=2, **kwargs):
+        if self.__class__ is DelayBase:
+            raise TypeError(('DelayBase must be subclassed with '
+                             'a "motor" component, the real motor to move.'))
+        self.n_bounces = n_bounces
+        super().__init__(*args, egu=egu, **kwargs)
+
+    @pseudo_position_argument
+    def forward(self, pseudo_pos):
+        """
+        Convert delay unit to motor unit
+        """
+        seconds = convert_unit(pseudo_pos.delay, self.delay.egu, 'seconds')
+        meters = seconds * speed_of_light / self.n_bounces
+        motor_value = convert_unit(meters, 'meters', self.motor.egu)
+        return self.RealPosition(motor=motor_value)
+
+    @real_position_argument
+    def inverse(self, real_pos):
+        """
+        Convert motor unit to delay unit
+        """
+        meters = convert_unit(real_pos.motor, self.motor.egu, 'meters')
```

Comment:
Clever

Reply:
The approach here was to protect against people using motors with random scales